src/SandboxingLevels.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Class SandboxingLevels.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AMP_Comments_Sanitizer;
+use AMP_Form_Sanitizer;
+use AMP_Options_Manager;
+use AMP_Script_Sanitizer;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Experimental service to facilitate flexible AMP.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2
+ * @internal
+ */
+final class SandboxingLevels implements Service, Registerable, Conditional {
+
+	/**
+	 * Option key for sandboxing level.
+	 *
+	 * @todo Move this to the Options interface once no longer experimental.
+	 * @var string
+	 */
+	const OPTION_SANDBOXING_LEVEL = 'sandboxing_level';
+
+	/**
+	 * Sandboxing levels.
+	 *
+	 * @var int[]
+	 */
+	const SANDBOXING_LEVELS = [ 1, 2, 3 ];
+
+	/**
+	 * Default sandboxing level.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_SANDBOXING_LEVEL = 3;
+
+	/**
+	 * Whether service is needed.
+	 *
+	 * @return bool
+	 */
+	public static function is_needed() {
+		/**
+		 * Filters whether experimental sandboxing is enabled.
+		 *
+		 * @internal
+		 * @since 2.2
+		 *
+		 * @param bool $enabled Sandboxing enabled.
+		 */
+		return (bool) apply_filters( 'amp_experimental_sandboxing_enabled', false );
```

Comment:
What happens with this filter if the experiment is stable and is the new default behavior? The name is temporary, so what is the deprecation strategy for names like this?